src/rectangle.js

```diff
@@ -223,67 +248,79 @@ $.Rect.prototype = /** @lends OpenSeadragon.Rect.prototype */{
      */
     // ----------
     union: function(rect) {
+        if (this.degrees !== 0 || rect.degrees !== 0) {
```

Comment:
It's kind of unfortunate to throw an exception here; I wonder if there might be a better way to fail. Perhaps return an empty rect and throw an error in the console? 

Of course you could also at least approximately solve the union for rotated rectangles. If they're both rotated the same degree, it should be fairly straightforward. If not the same degree, you could return a horizontal rectangle that is the union of the boundingBox for each.